testdata/dnn/onnx/generate_onnx_models.py

```diff
@@ -1916,37 +1973,6 @@ def forward(self, x):
 model = GatherMultiOutput()
 save_data_and_model("gather_multi_output", x, model)
 
-def postprocess_model(model_path, inputs_shapes):
-    onnx_model = onnx.load(model_path)
-
-    def update_inputs_dims(model, input_dims):
-        """
-            This function updates the sizes of dimensions of the model's inputs to the values
-            provided in input_dims. if the dim value provided is negative, a unique dim_param
-            will be set for that dimension.
-        """
-        def update_dim(tensor, dim, i, j, dim_param_prefix):
-            dim_proto = tensor.type.tensor_type.shape.dim[j]
-            if isinstance(dim, int):
-                if dim >= 0:
-                    dim_proto.dim_value = dim
-                else:
-                    dim_proto.dim_param = dim_param_prefix + str(i) + '_' + str(j)
-            elif isinstance(dim, str):
-                dim_proto.dim_param = dim
-            else:
-                raise ValueError('Only int or str is accepted as dimension value, incorrect type: {}'.format(type(dim)))
-
-        for i, input_dim_arr in enumerate(input_dims):
-            for j, dim in enumerate(input_dim_arr):
-                update_dim(model.graph.input[i], dim, i, j, 'in_')
-
-        onnx.checker.check_model(model)
-        return model
-
-    onnx_model = update_inputs_dims(onnx_model, inputs_shapes)
-    onnx.save(onnx_model, model_path)
```

Comment:
Not sure it it should be moved

Reply:
Otherwise it fails with this error [#23508] (https://github.com/opencv/opencv/issues/23508)

Reply:
Need to merge https://github.com/opencv/opencv_extra/pull/1059 first then

Reply:
Done.